src/pages/VideoDetail/VideoDetail.js

```diff
@@ -0,0 +1,147 @@
+import React, { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { BiLike, BiPlayCircle, BiTime } from "react-icons/bi";
+import { MdOutlineWatchLater, MdWatchLater } from "react-icons/md";
+import { AiOutlineEye, AiFillLike } from "react-icons/ai";
+import { PlaylistModal, Sidebar } from "components";
+import { useAuth, useLike, useWatchLater } from "context";
+import {
+  likeVideo,
+  unLikeVideo,
+  addToWatchLater,
+  removeFromWatchLater,
+} from "services";
+import axios from "axios";
+import "./VideoDetail.css";
+
+export const VideoDetail = () => {
+  const [singleVideoDetail, setSingleVideoDetail] = useState(null);
+  const [isLiked, setIsLiked] = useState(false);
+  const [isWatchLater, setIsWatchLater] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+
+  const { likedVideos, setLikedVideos } = useLike();
+  const { watchLaterVideos, setWatchLaterVideos } = useWatchLater();
+
+  const { auth } = useAuth();
+  const { videoId } = useParams();
+  const navigate = useNavigate();
+  const { title, views, description, thumbnail, duration, _id, creator } =
+    singleVideoDetail ?? {};
+
+  useEffect(() => {
+    setIsLiked(likedVideos.find((video) => video._id === videoId));
```

Comment:
You can create a utility function to check this condition, as you are using it for other features also.